src/lycophron/project.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Lycophron is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""Lycophron project classes (business logic layer)."""
+
+import csv
+import os
+
+from .errors import (
+    DatabaseAlreadyExists,
+    ErrorHandler,
+    InvalidRecordData,
+)
+from .loaders import LoaderFactory
+from .schemas.record import RecordRow
+
+
+class Project:
+    def __init__(self, project_folder=os.getcwd()):
+        self.project_folder = project_folder
+        self.errors = []
+
+    def load_file(self, filename):
+        data = self.process_file(filename)
+        for record in data:
+            self.add_record(record)
+        if len(self.errors):
+            ErrorHandler.handle_error(self.errors)
+
+    def process_file(self, filename):
+        if not self.is_project_initialized():
+            raise Exception("Project is not initialised.")
+        factory = LoaderFactory()
```

Comment:
using loader factory, tested and the behavior is the same as before. However, if the file format is not supported an exception is raised. Probably we want to expose that message to the user (e.g. "`.xslx` is not supported yet")